commands/verify/verify.go

```diff
@@ -87,6 +88,87 @@ func verifySingleCase(expectedFile, actualFile, query string) error {
 	return nil
 }
 
+// concurrentErrors store errors that occurred when verifying cases in goroutines.
+type concurrentErrors struct {
+	errs  *multierror.Error
+	mutex sync.Mutex
+}
+
+// verifyInfo contains necessary information about verification
+type verifyInfo struct {
+	retryCount int
+	interval   time.Duration
+	failFast   bool
+}
+
+func concurrentSafeErrAppend(concurrentError *concurrentErrors, err error) {
+	concurrentError.mutex.Lock()
+	concurrentError.errs = multierror.Append(concurrentError.errs, err)
+	concurrentError.mutex.Unlock()
+}
+
+func check(stopChan chan bool, goroutineNum int) bool {
+	count := 0
+	for {
+		v, ok := <-stopChan
+		if ok {
+			if !v {
+				return false
+			}
+
+			count++
+		}
+
+		if count == goroutineNum {
+			return true
+		}
+	}
+}
+
+func concurrentVerifySingleCase(idx int, v config.VerifyCase, errs *concurrentErrors, verify verifyInfo, wg *sync.WaitGroup, stopChan chan bool) {
+	var err error
+
+	defer func() {
+		if verify.failFast {
+			if err != nil {
+				concurrentSafeErrAppend(errs, err)
+				stopChan <- false
+			} else {
+				stopChan <- true
```

Comment:
It's confusing here, because when we send `false` to `stopChan`, the program should exit.